server/src/analytics/views/activity.ts

```diff
@@ -0,0 +1,55 @@
+import { Activity } from '@prisma/client';
+import { PrismaService } from 'src/prisma.service';
+
+export interface ActivityViewSupportedFields {
+  count?: boolean;
+  revenue?: boolean;
+}
+
+export interface ActivityView extends Activity {
+  count?: number;
+  revenue?: number;
+}
+
+export class ActivityDataView {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async get({ count, revenue }: ActivityViewSupportedFields) {
+    let baseData = await this.prisma.activity.findMany({
+      where: {
+        isArchived: false,
+      },
+    });
+    if (count) {
+      baseData = await this.addCountData(baseData);
+    }
+    if (revenue) {
+      baseData = await this.addRevenueData(baseData);
+    }
+    return baseData;
+  }
+
+  private async addRevenueData(data: Activity[]) {
+    const newData = await this.addCountData(data);
+    return newData.map((a) => ({
```

Comment:
whats `a`?

Reply:
should be activity, copilot wrote this I thought it made sense in the context but I will improve the variable names